server/auth/providers/email.js

```diff
@@ -84,25 +84,26 @@ router.post("email", async (ctx) => {
   };
 });
 
-router.get("email.callback", auth({ required: false }), async (ctx) => {
+router.get("email.callback", async (ctx) => {
   const { token } = ctx.request.query;
 
   ctx.assertPresent(token, "token is required");
 
   try {
     const user = await getUserForEmailSigninToken(token);
-
-    const team = await Team.findByPk(user.teamId);
-    if (!team.guestSignin) {
-      throw new AuthorizationError();
+    if (!user.team.guestSignin) {
+      return ctx.redirect("/?notice=auth-error");
+    }
+    if (user.isSuspended) {
```

Comment:
We need to check and redirect _before_ marking the user as active, previously this was done inside of `signIn`